queries/soql/highlights.scm

```diff
@@ -82,9 +82,9 @@
 
 (string_literal) @string
 
-(date) @variable.readonly
+(date) @variable
 
-(date_time) @variable.readonly
+(date_time) @variable
```

Comment:
Grouping these 2 captures maybe

Reply:
Should these be `@string.special` instead?

Reply:
Seems like that https://github.com/aheber/tree-sitter-sfapex/blob/857077f9e6bb04df0f769c18d32bfe036911adc8/common/soql-grammar.js#L415

But I don't think this PR should remove the fallback highlight (if possible).

Reply:
> But I don't think this PR should remove the fallback highlight (if possible).

What do you mean?

Reply:
As in, I think this PR should just remove the undocumented captures and use fallback if possible. Of course if the fallback no longer fits the usage criteria then we have to use another capture. 

But for this, I think it still fits, and will rather let the grammar maintainers to decide.

Reply:
Nevermind, I second-thought and would rather you to switch it to string.special :D

Doesn't make sense for a datestring to be a variable

Reply:
So which is it? Adapt to the right usage, or not? ;)